### PR TITLE
Refactor and improve etcdexpose

### PR DESCRIPTION
**Pull request by jlevesy**
This is an initial refactor of etcdexpose.
- Moved to the new 'etcd/client' library instead of the deprecated go-etcd
- Handlers are executed asynchronously
- All channels involved are now buffered, which was causing the deadlock issue
- Reorganize sources in a more "go-ish" way (still work to do)
- Introduce some tests

There is still work to do, as we might need to blow up some useless abstractions and improve our test suite.
We also need to setup some vendoring.

Deals with https://github.com/upfluence/man/issues/60

@AlexisMontagne @phndiaye what do you think ?

### CODE REVIEW

File: handler/single/handler.go [hunk 1]
```diff
@@ -1,21 +1,23 @@
-package etcdexpose
+package single
 
 import (
+	"github.com/coreos/etcd/client"
```
Comment (AlexisMontagne) [inline handler/single/handler.go:4]: the abstraction is kinda broken if we use explicitly etcd values

Comment (jlevesy) [inline handler/single/handler.go:4]: Yeah, I know, I wanted to ship fast, then removing that crap later.

**Comment by AlexisMontagne** [comment]
Could you split the `utils` package, and create an `healthcheck` and `value_renderer` package, it would be more sustainable in the long run

**Comment by AlexisMontagne** [comment]
Integrate CircleCI \o/

**Comment by AlexisMontagne** [comment]
`handler/single` and `handler/multiple` are 95% the same, there is any way to just create one `Handler` struct, and  create two implementation of a `Renderer` or `Picker` such as:

``` golang
 type Picker {
    Render(nodes client.Nodes) (string, error)
}
```

**Comment by jlevesy** [comment]
I was thinking about something like that, I said I wanted to ship fast then improving it :)

**Pull request merged**